aya/src/bpf.rs

```diff
@@ -481,11 +484,7 @@ impl Bpf {
     /// # Ok::<(), aya::BpfError>(())
     /// ```
     pub fn program(&self, name: &str) -> Result<&Program, ProgramError> {
-        self.programs
-            .get(name)
-            .ok_or_else(|| ProgramError::NotFound {
-                name: name.to_owned(),
-            })
+        self.programs.get(name).ok_or(ProgramError::NotFound)
```

Comment:
I agree, and this was actually my initial version.

I could however potentially see this being slightly less ergonomic as the API user is forced to .ok_or() themself if they want to use this in a function returning a Result<T, Error>, and they must create their own error type.

In another PR, I add program_opt / program_opt_mut to explicitly provide Option<T> versions of these functions.

Let me know if you prefer that approach.

Reply:
>  I could however potentially see this being slightly less ergonomic as the API user is forced to .ok_or() themself if they want to use this in a function returning a Result<T, Error>, and they must create their own error type.

They'll have to `ok_or()` for any other rust functions that return `Option`, so I think it's fine

Reply:
Okay, I think that this latest push has addressed this and all of the documentation surrounding it.

Reply:
Omg, I'm so sorry I just realised that I was wrong 😭. 

Now I remember why I made it return `ProgramError::NotFound`: the reason is that you effectively never call `program()` and `program_mut()` without calling `try_into()` after, which return a `Result`, and so it becomes kinda overwhelming having to do `bpf.program_mut("foo").ok_or(...).try_into()?` every time.

Could you please undo this? I owe you one Thia 😅

Reply:
I think that this is why git has the reflog.

Reply:
Actually after writing that I realised that it's probably fine to unwrap after `program()` and `program_mut()` since the name is going to be static 99% of the times and so it's probably a typo if the result isn't Some. Oh well, my brain obviously keeps flip flopping about this so i'll leave it to you :D

Reply:
I may have gotten whiplash from this.

For what it's worth, in our daemon, we're doing `.program_opt().unwrap_unchecked()`.

I am in favor of the Option return. I'll revert again...